src/test/java/teammates/test/cases/testdriver/TestngTest.java

```diff
@@ -0,0 +1,138 @@
+package teammates.test.cases.testdriver;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import org.testng.annotations.Test;
+
+import teammates.test.cases.BaseTestCase;
+
+public class TestngTest extends BaseTestCase {
+
+    private Scanner scanner;
+    List<String> tests = new ArrayList<String>();
+    List<String> directoriesTested = new ArrayList<String>();
+
+    @Test
+    public void checksTestsInTestNg() {
+        
+        // Verify that testng.xml exists
+        File f = new File("./src/test/testng.xml");   
+        assertTrue(f.exists() && !f.isDirectory());
+        
+        updateDirectoriesTested();
+        getTestFiles("./src/test/java/teammates/test");
+        excludeFilesNotInTestng();
+
+        for (int i = 0; i < tests.size(); i++) {
+            verifyTestngContainsTest(f, tests.get(i));
+        }
+    }
+
+    /**
+     * Get all Java test files from directories which are included in directories tested
+     * @param path  path to search in
+     */
+    private void getTestFiles(String path) {
+        File folder = new File(path);
+        File[] listOfFiles = folder.listFiles();      
+
+        for (int i = 0; i < listOfFiles.length; i++) {
+            String name = listOfFiles[i].getName();
+            
+            if (listOfFiles[i].isFile() && name.endsWith(".java")) {
+                tests.add(name.replace(".java", ""));
+                
+            } else if (listOfFiles[i].isDirectory()) {
+            
+                if (directoriesTested.contains(name)) {
+                    getTestFiles(path + "/" + name);
+                } else if (containsNestedDirectory(name)) {
+                    
+                    List<String> nestedDirs = getAllNestedDirectories(name);
+                    
+                    for (int j = 0; j < nestedDirs.size(); j++) {
+                        getTestFiles(path + "/" + nestedDirs.get(j));
+                    }
+                }
+            }
+        }
+    }
```

Comment:
As far as possible, try to avoid manipulating global variables. This method should return the list of files instead of modifying `tests`. Furthermore, this is a general method that can be moved to a utility class.